pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt

```diff
@@ -1,9 +1,36 @@
 package com.pubnub.api.endpoints.push
 
+import cocoapods.PubNubSwift.PubNubObjC
 import com.pubnub.api.Endpoint
+import com.pubnub.api.PubNubException
+import com.pubnub.api.enums.PNPushType
 import com.pubnub.api.models.consumer.push.PNPushRemoveAllChannelsResult
+import com.pubnub.api.onFailureHandler
+import com.pubnub.api.v2.callbacks.Consumer
+import com.pubnub.api.v2.callbacks.Result
+import com.pubnub.kmp.toNSData
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.Foundation.NSData
 
 /**
  * @see [PubNub.removeAllPushNotificationsFromDeviceWithPushToken]
  */
-actual interface RemoveAllPushChannelsForDevice : Endpoint<PNPushRemoveAllChannelsResult>
\ No newline at end of file
+actual interface RemoveAllPushChannelsForDevice : Endpoint<PNPushRemoveAllChannelsResult>
+
+@OptIn(ExperimentalForeignApi::class)
+open class RemoveAllPushChannelsForDeviceImpl(
```

Comment:
I am wondering if "open" is required here?

Reply:
good point, it's probably not required?

Reply:
I'll check it and remove all occurrences I've written so far in the next PR if it turns out that `open` is not required